lib/java/com/google/android/material/floatingactionbutton/ExtendedFloatingActionButton.java

```diff
@@ -217,6 +234,21 @@ protected void onAttachedToWindow() {
     return behavior;
   }
 
+  /**
+   * Determines how the extended button is hidden when it needs to be collapsed
+   * @param isHide If true, none of the button will be visible when collapsed. If false, the button
+   * will "shrink" and hide button text but keep the icon.
+   *
+   * This option is "false" and perform a shrink by default.
+   */
+  public void setHideBehavior(boolean isHide) {
```

Comment:
Because of the addition of this method, it looks like this api is a bit confusing. I think because if `autoShrink` is true it takes precedence over `autoHide` we could make this into an IntDef or something where you can only set the collapse behavior to one of them. For example, you could call `setCollapseBehavior(HIDE)` to perform the same thing as `setHideBehavior(true)`, or `setCollapseBehavior(SHRINK)` for `setHideBehavior(false)`. Then you'd also have any easy way to disable any of this by calling `setCollapseBehavior(NONE)`. What do you think about adding that method and deprecating the others?

Reply:
That it what I invisioned, for an enumaration. Originally I just wanted to keep my changes minimal was to maintain the current API. 

Great, glad you are onboard with the changesm I will change the booleans to enums in the client method and throughout the behavior class

Reply:
I don't know how realistic this is but is there absolutely no way to combine both? Does it have to be one OR the other? Why not? I see a possible need to first autoshrink the button when collapsing the app bar to the regular/thin toolbar and then autohiding altogether when the toolbar is also collapsed. It seems to me there's a scenario when you might want both.

Reply:
@falsevapor You may be on to something, how about you post that in a separate GitHub issue? I will address that when this is complete 😉 

With this PR, we have to address some challenges integrating the ExtFABBehavior class. This PR does not to be more complicated than it already is.

Reply:
@cketcham What do you think about this decision. Should we make the ExtFABBehavior class public, or keep it how it is and expose a client method?

Reply:
right, for now just to allow the access to one or the other is enough to get my original question answered/handled.

Reply:
Ok

Reply:
Thank you. Just let me know when you are ready for me to test this. Seems like it's still in discussion.

Reply:
I think having the option to first auto shrink and then hide would be great! But it's also outside the scope of this PR. Maybe we could change the api that you've added to `ExtendedFloatingActionButton`? Instead of having the method take a boolean, you could have it take an intdef or enum for HIDE, SHRINK, and NONE.

In the future, this should allow us to add new enum values that let us create more complicated behavior such as the option to do both like you've described.

Reply:
Agreed. I opened a new ticket about allowing both.